cli/src/complete.rs

```diff
@@ -837,10 +856,18 @@ mod parse {
     // the files changed only in some other revision in the range between
     // --from and --to cannot be squashed into --to like that.
     pub fn squash_revision() -> Option<String> {
-        if let Some(rev) = parse_flag(&["-r", "--revision"], std::env::args()) {
+        if let Some(rev) = parse_flag(&["-r", "--revision"], &mut std::env::args()) {
             return Some(rev);
         }
-        parse_flag(&["-f", "--from"], std::env::args())
+        parse_flag(&["-f", "--from"], &mut std::env::args())
+    }
+
+    // Special parse function only for `jj log`. It has a --revisions flag,
+    // instead of the usual --revision, and it can be supplied multiple times.
+    pub fn log_revisions() -> Vec<String> {
+        let candidates = &["-r", "--revisions"];
+        let mut args = std::env::args();
+        std::iter::from_fn(|| parse_flag(candidates, &mut args)).collect()
```

Comment:
nit: I would make `parse_flag()` return an iterator. It's not so obvious where the input iterator points after `parse_flag()` returned `Some(_)`. It's also easier to write a unit test.